src/Bedrock.Framework.Experimental/Protocols/Kafka/StrategyPayloadWriter.cs

```diff
@@ -0,0 +1,264 @@
+﻿#nullable enable
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Bedrock.Framework.Experimental.Protocols.Kafka
+{
+    public class BigEndianStrategy : IPayloadWriterStrategy
```

Comment:
If performance is important, make these structs, and make the classes that call them generic over the interface.